### DISCUSSION TITLE
Consider switching away from modernc.org/sqlite driver

### DISCUSSION THREAD

**Discussion by gotmax23** [aquasecurity/trivy#6449] [Q&A]
During the [Fedora Linux package review for trivy](https://bugzilla.redhat.com/show_bug.cgi?id=2272258), we found some potential licensing issues with the modernc components. See https://gitlab.com/cznic/libc/-/issues/31. AFAIU, the issue is the large amounts of generated code derived from GPL-licensed glibc header files. Would trivy be willing to consider a different sqlite backend such as github.com/mattn/go-sqlite3 to avoid this issue?

**Comment by knqyf263** [comment]
We're open to switching a library, but we need a CGO-free library.

**Reply by gotmax23** [reply]
What's wrong with CGO? You can still create static binaries with musl[^1] if that's the blocker.

[^1]: Something like `CC=musl-gcc go build -v -ldflags '-linkmode external -extldflags "-static"'`; see https://honnef.co/articles/statically-compiled-go-programs-always-even-with-cgo-using-musl/

**Reply by gotmax23** [reply]
I say this, because most of the Go sqlite drivers do link to libsqlite with CGO. This seems like a much better approach than doing strange things with transpiling and generated code.

**Reply by knqyf263** [reply]
CGO makes cross-compilation much harder. We used to use CGO and were suffering from that. Then, we have removed CGO with great effort.

**Reply by gotmax23** [reply]
Fair enough. For now, I think we can patch it out ourselves downstream, so we can use CGO and link to system libraries—like we generally prefer—and avoid potential licensing issues.

**Reply by gotmax23** [reply]
Thank you for your help and for explaining your position.
